firmware/applications/SignBuddy/adc.c

```diff
@@ -0,0 +1,95 @@
+#include "adc.h"
+
+#include "board.h"
+
+#define ADC_PREEMPT_PRIO                     0
+#define ADC_SUB_PRIO                         0
+
+#define VREF                                 ((uint32_t)5000)
+#define ADC_DELAY_CALIB_ENABLE_CPU_CYCLES    (LL_ADC_DELAY_CALIB_ENABLE_ADC_CYCLES * 32)
+#define VAR_CONVERTED_DATA_INIT_VALUE        (__LL_ADC_DIGITAL_SCALE(LL_ADC_RESOLUTION_12B) + 1)
+
+void adc_init(void)
+{
+  LL_ADC_InitTypeDef adc_config = { 0 };
+  LL_ADC_REG_InitTypeDef adc_reg_config = { 0 };
+  LL_ADC_CommonInitTypeDef adc_common_config = { 0 };
+
+  LL_GPIO_InitTypeDef GPIO_InitStruct = { 0 };
+
+  GPIO_ADC_CLK_EN();
+
+  GPIO_InitStruct.Pin = ADC_PIN;
+  GPIO_InitStruct.Mode = LL_GPIO_MODE_ANALOG;
+  GPIO_InitStruct.Pull = LL_GPIO_PULL_NO;
+  LL_GPIO_Init(ADC_PORT, &GPIO_InitStruct);
+
+  LL_GPIO_EnablePinAnalogControl(ADC_PORT, ADC_PIN);
+
+  NVIC_SetPriority(ADC1_2_IRQn, NVIC_EncodePriority(NVIC_GetPriorityGrouping(), ADC_PREEMPT_PRIO, ADC_SUB_PRIO));
```

Comment:
add a define for the irq too?

Reply:
See other reply, you would also need to abstract the IRQ callback, not sure how to do that